tests/simulator/test_initialization.py

```diff
@@ -2,95 +2,117 @@
 from pyphare.cpp import cpp_lib
 cpp = cpp_lib()
 
-from pyphare.simulator.simulator import Simulator, startMPI
+from pyphare.simulator.simulator import Simulator
 from pyphare.pharesee.hierarchy import hierarchy_from, merge_particles
-from pyphare.pharein import MaxwellianFluidModel
+from pyphare.pharein import MaxwellianFluidModel, fn_wrapper
 from pyphare.pharein.diagnostics import ParticleDiagnostics, FluidDiagnostics, ElectromagDiagnostics
 from pyphare.pharein import ElectronModel
 from pyphare.pharein.simulation import Simulation
-from pyphare.pharesee.geometry import level_ghost_boxes, hierarchy_overlaps
+from pyphare.pharesee.geometry import level_ghost_boxes, hierarchy_overlaps, touch_domain_border
 from pyphare.pharesee.particles import aggregate as aggregate_particles
 import pyphare.core.box as boxm
-from pyphare.core.box import Box, Box1D
+from pyphare.core.box import Box, Box1D, Box2D, Box3D, nDBox
 import numpy as np
 import unittest
 from ddt import ddt, data, unpack
 
 
+from datetime import datetime
 
 
 @ddt
 class InitializationTest(unittest.TestCase):
 
+    def datetime_now(self):
+        return datetime.now()
+
+    def datetime_diff(self, then):
+        return (datetime.now() - then).total_seconds()
+
+
     def ddt_test_id(self):
         return self._testMethodName.split("_")[-1]
 
-    def getHierarchy(self, interp_order, refinement_boxes, qty, nbr_part_per_cell=100,
-                     diag_outputs="phare_outputs",
-                     density = lambda x: 0.3 + 1./np.cosh((x-6)/4.)**2,
-                     beam = False, time_step_nbr=1,
-                     smallest_patch_size=10, largest_patch_size=10,
-                     cells= 120,
-                     dl=0.1):
+    def _density(*xyz):
+        x = xyz[0]
+        return 0.3 + 1./np.cosh((x-6)/4.)**2
```

Comment:
we should probably consider a 2D function in 2D (i.e. that depends on both x and y).
I suggest : 

nb + np.exp(-(x-x0)**2-(x-x0)**2)

where nb is for instance 1
and x0 and y0 are the middle of the box in the associated direction.

Reply:
still todo